demos/human_pose_estimation_demo/src/human_pose_estimator.cpp

```diff
@@ -38,14 +38,63 @@ HumanPoseEstimator::HumanPoseEstimator(const std::string& modelPath,
     std::string binFileName = fileNameNoExt(modelPath) + ".bin";
     netReader.ReadWeights(binFileName);
     network = netReader.getNetwork();
-    InferenceEngine::InputInfo::Ptr inputInfo = network.getInputsInfo().begin()->second;
-    inputLayerSize = cv::Size(inputInfo->getTensorDesc().getDims()[3], inputInfo->getTensorDesc().getDims()[2]);
-    inputInfo->setPrecision(InferenceEngine::Precision::U8);
+
+    const auto& inputInfo = network.getInputsInfo();
+
+    if (inputInfo.size() != 1) {
+        throw std::runtime_error(modelPath + ": expected to have 1 input");
+    }
+
+    const auto& imageInputInfo = *inputInfo.begin();
+    const auto& imageInputDims = imageInputInfo.second->getTensorDesc().getDims();
+
+    if (imageInputDims.size() != 4 || imageInputDims[0] != 1 || imageInputDims[1] != 3) {
+        throw std::runtime_error(
+            modelPath + ": expected \"" + imageInputInfo.first + "\" to have dimensions 1x3xHxW");
+    }
+
+    inputLayerSize = cv::Size(imageInputDims[3], imageInputDims[2]);
+    imageInputInfo.second->setPrecision(InferenceEngine::Precision::U8);
 
     InferenceEngine::OutputsDataMap outputInfo = network.getOutputsInfo();
-    auto outputBlobsIt = outputInfo.begin();
-    pafsBlobName = outputBlobsIt->first;
-    heatmapsBlobName = (++outputBlobsIt)->first;
+
+    if (outputInfo.size() != 2) {
+        throw std::runtime_error(modelPath + ": expected to have 2 outputs");
+    }
+
+    auto outputIt = outputInfo.begin();
+
+    const auto& pafsOutputInfo = *outputIt++;
+
+    pafsBlobName = pafsOutputInfo.first;
+
+    const auto& pafsOutputDims = pafsOutputInfo.second->getTensorDesc().getDims();
+
+    if (pafsOutputDims.size() != 4 || pafsOutputDims[0] != 1
+            || pafsOutputDims[1] != 2 * (keypointsNumber + 1)) {
+        throw std::runtime_error(
+            modelPath + ": expected \"" + pafsBlobName + "\" to have dimensions "
+                "1x" + std::to_string(2 * (keypointsNumber + 1)) + "xHFMxWFM");
+    }
+
+    const auto& heatmapsOutputInfo = *outputIt++;
```

Comment:
```suggestion
    const auto& heatmapsOutputInfo = *outputIt;
```
Why is `outputIt` incremented?

Reply:
It pleases me to have the iterator reach the end of the container by the end of the function. :-)

Reply:
OK :)